### ISSUE TITLE
Semi-Global Alignment

### ISSUE THREAD

**Issue by mozack** [vgteam/vg#1936]
Hello,

It looks like VG supports either local or global alignment.  Is there currently a way to do semi-global alignment?  If not, could it be added?

Thanks!

**Comment by adamnovak** [comment]
What exactly are the semantics of the semi-global alignment that you want? Is it aligning a sequence to a graph where gaps at the ends of the query sequence are free, but without soft-clipping of the query sequence?

**Comment by mozack** [comment]
I'm looking for end-to-end alignment of the query sequence (including penalties for gaps at the ends) but localized alignment in the context of the graph.  i.e. Gaps beginning from the source or ending at the sink would be free.

**Comment by adamnovak** [comment]
That sounds like what I am thinking of. Gaps characters in the query
(i.e. unused graph bases) are free before the first base or after the
last base of the query string, while gap characters in the reference
side of the alignment (i.e. unused search sequence bases) are
penalized.

I think we could do this in `vg align`, by turning off soft clipping,
but I don't think there's an option for it right now.

**Comment by mozack** [comment]
Yes, what you've described is exactly what I would be interested in.  If it is possible to make this an option in a future release, that would be great.  Thanks!

**Comment by pesho-ivanov** [comment]
Is there any progress on this feature? I am looking to compare the exact vg align with other semi-global aligners.

**Comment by adamnovak** [comment]
I don't think anyone has worked on it yet, unfortunately. I also don't think a workaround of a positive score for a mismatch would work, either; we pass mismatch scores around internally as unsigned, and force them to be negative.

**Comment by ekg** [comment]
Does xdrop meet your requirements? It's described in the dozeu repository
and enabled optionally in vg map.
